### PR TITLE
fix 5 set related bugs reported by Mercury

### PR DESCRIPTION
**Pull request by rubio41**
- [x] 1. at 9am every Sunday, should be contacted
- [x] 2. every morning at 9am, the timex is TMO, should merge morning and 9am
- [x] 3. schedule a repeating event on mondays and wednesdays, this should not be a range, but two separate sets (related with the monday+s bug)
- [x] 4. on Mondays = every Monday
- [x] 5. 7 til 8am = 7 until 8am

### CODE REVIEW

File: Microsoft.Recognizers.Text.DateTime/Spanish/Extractors/SpanishDatePeriodExtractorConfiguration.cs [hunk 1]
```diff
@@ -72,7 +72,7 @@ public class SpanishDatePeriodExtractorConfiguration : IDatePeriodExtractorConfi
         //TODO: modify it according to the related regex in English
         public static readonly Regex OneWordPeriodRegex =
             new Regex(
-                @"\b(((pr[oó]xim[oa]?|est[ea]|[uú]ltim[oa]?|en)\s+)?(?<month>Abril|Abr|Agosto|Ago|Diciembre|Dic|Enero|Ene|Febrero|Feb|Julio|Jul|Junio|Jun|Marzo|Mar|Mayo|May|Noviembre|Nov|Octubre|Oct|Septiembre|Setiembre|Sept|Set)|(?<=\b(del|de la|el|la)\s+)?(pr[oó]xim[oa](s)?|[uú]ltim[oa]?|est(e|a))?\s+(fin de semana|semana|mes|año)|fin de semana|(mes|años)? a la fecha)\b",
+                @"\b(((pr[oó]xim[oa]?|est[ea]|[uú]ltim[oa]?|en)\s+)?(?<month>Abril|Abr|Agosto|Ago|Diciembre|Dic|Enero|Ene|Febrero|Feb|Julio|Jul|Junio|Jun|Marzo|Mar|Mayo|May|Noviembre|Nov|Octubre|Oct|Septiembre|Setiembre|Sept|Set)|(?<=\b(del|de la|el|la)\s+)?(pr[oó]xim[oa](s)?|[uú]ltim[oa]?|est(e|a))\s+(fin de semana|semana|mes|año)|fin de semana|(mes|años)? a la fecha)\b",
```
Comment (tellarin) [inline Microsoft.Recognizers.Text.DateTime/Spanish/Extractors/SpanishDatePeriodExtractorConfiguration.cs]: Why this change?

File: Patterns/English/English-DateTime.yaml [hunk 2]
```diff
@@ -18,7 +18,7 @@ MonthNumRegex: !simpleRegex
 PeriodYearRegex: !simpleRegex
   def: \b(?<year>19\d{2}|20\d{2})\b
 WeekDayRegex: !simpleRegex
-  def: (?<weekday>Sunday|Monday|Tuesday|Wednesday|Thursday|Friday|Saturday|Mon|Tue|Wedn|Weds|Wed|Thurs|Thur|Thu|Fri|Sat|Sun)\b
+  def: \b(?<weekday>Sunday|Monday|Tuesday|Wednesday|Thursday|Friday|Saturday|Mon|Tue|Wedn|Weds|Wed|Thurs|Thur|Thu|Fri)\b
```
Comment (tellarin) [inline Patterns/English/English-DateTime.yaml]: Why remove Sat and Sun?

File: Patterns/English/English-DateTime.yaml [hunk 3]
```diff
@@ -297,7 +297,9 @@ EachUnitRegex: !nestedRegex
   def: (?<each>(each|every)\s*{DurationUnitRegex})
   references: [ DurationUnitRegex ]
 EachPrefixRegex: !simpleRegex
-  def: (?<each>(each|every)\s*$)
+  def: \b(?<each>(each|(every(\s+other)?))\s*$)
```
Comment (tellarin) [inline Patterns/English/English-DateTime.yaml]: Where do you deal with the 'every other' case in resolution?

File: Microsoft.Recognizers.Text.DateTime/Spanish/Extractors/SpanishDatePeriodExtractorConfiguration.cs (hunk 1 above)
Comment (rubio41) [reply Microsoft.Recognizers.Text.DateTime/Spanish/Extractors/SpanishDatePeriodExtractorConfiguration.cs]: In English, this pattern is used to extract (next|past|...) (month|week|day...). We didn't want to extract single "month|week|day" but require a prefix. I made the changes in English, but not in Spanish. In addition, for the single "month|week|day", this pattern will output an extraction start with "\s" which is not expected.

File: Patterns/English/English-DateTime.yaml (hunk 2 above)
Comment (rubio41) [reply Patterns/English/English-DateTime.yaml]: "Sat" and "Sun" are ambiguous words. It is not correct to extract that alone. I use a new regex for the SingleWeekdayRegex which deleted the Sat and Sun.

File: Patterns/English/English-DateTime.yaml (hunk 3 above)
Comment (rubio41) [reply Patterns/English/English-DateTime.yaml]: The original plan is to add it later. Now, I treat every other week = biweekly.

File: Microsoft.Recognizers.Text.DateTime/Spanish/Extractors/SpanishDatePeriodExtractorConfiguration.cs (hunk 1 above)
Comment (tellarin) [reply Microsoft.Recognizers.Text.DateTime/Spanish/Extractors/SpanishDatePeriodExtractorConfiguration.cs]: OK
